modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EChain.java

```diff
@@ -249,6 +249,10 @@ private void analyzeRead() {
     @Override
     void write(MethodWriter writer) {
         if (cat) {
+            writer.writeDebugInfo(offset);
+        }
+
+        if (cat) {
             writer.writeNewStrings();
```

Comment:
this node has more `if (cat)` cases beneath and is kinda complex.  I think maybe too complex for bytecode manipulation in a single node? Or at least it needs some assistance. If it gets refactored, i didnt want the debug info to go away with it on accident...